### PR TITLE
Add localized idiomatic deictic expression for zh_CN

### PR DESCRIPTION
**Pull request by Elethom**

### CODE REVIEW

**Comment by yas375** [comment]
@Elethom thanks for the pull request! Would you mind to rebase this on master and update/add some unit tests for this? Or do you want me to do this?

**Comment by Elethom** [comment]
@yas375 Done. 😃

**Comment by yas375** [comment]
@Elethom awesome!

I can see CI reports some failed tests:
```
 testIdiomaticPast, XCTAssertEqual failed: ("Optional("昨天")") is not equal to ("Optional("1天前")") - 
  /Users/travis/build/mattt/FormatterKit/Example/Tests-Chinese-Traditional/TimeIntervalFormatterTests.swift:46

testIdiomaticPast, XCTAssertEqual failed: ("Optional("前天")") is not equal to ("Optional("2天前")") - 
  /Users/travis/build/mattt/FormatterKit/Example/Tests-Chinese-Traditional/TimeIntervalFormatterTests.swift:47
```

Will you have a chance to look into it? You can run these tests locally from Xcode. If you want to run tests for all localizations you can do this by calling `rake spec` from command line.

File: Example/Tests-Chinese-Simplified/TimeIntervalFormatterTests.swift [hunk 1]
```diff
@@ -25,7 +25,7 @@ class TTTTimeIntervalFormatterTests: XCTestCase {
         XCTAssertEqual(formatter.stringForTimeInterval(-10000), "2小时前")
         XCTAssertEqual(formatter.stringForTimeInterval(-100000), "1天前")
         XCTAssertEqual(formatter.stringForTimeInterval(-200000), "2天前")
-        XCTAssertEqual(formatter.stringForTimeInterval(-10000000), "3月前")
+        XCTAssertEqual(formatter.stringForTimeInterval(-10000000), "3个月前")
```
Comment (yas375) [inline Example/Tests-Chinese-Simplified/TimeIntervalFormatterTests.swift]: Thanks for the explanation about `月` and `个` in the commit description! 👍

**Comment by Elethom** [comment]
@yas375 That's weird. The tests passed on my Mac. Probably an issue with locale arguments. Would you mind checking it out for me? I will do it later if you couldn't. I'm out for now considering it's already 3:00 am here. 💤

**Comment by yas375** [comment]
@Elethom sure! I'll take a look now. Have some rest!

File: FormatterKit/TTTTimeIntervalFormatter.m [hunk 2]
```diff
@@ -270,6 +270,11 @@ - (NSString *)localizedIdiomaticDeicticExpressionForComponents:(NSDateComponents
         return [self plRelativeDateStringForComponents:components];
     } else if ([languageCode isEqualToString:@"hu"]) {
         return [self huRelativeDateStringForComponents:components];
+    } else if ([languageCode isEqualToString:@"zh"]) {
+        NSString *languageCountry = [self.locale objectForKey:NSLocaleCountryCode];
+        if ([languageCountry isEqualToString:@"CN"]) {
```
Comment (yas375) [inline FormatterKit/TTTTimeIntervalFormatter.m]: I've investigated unit tests failure. The changes in this pull request are intended to Chinese Simplified. Tests failed in Chinese Traditional target. If changes in `zhCNRelativeDateStringForComponents` are specific to Chinese Simplified, then we need to fix the way of targeting it.

I've tried to switch Languages and Regions in iOS Simulator and have recorded locale values:

```
1. English (Region: United States)

NSLocaleIdentifier en_US
NSLocaleLanguageCode en
NSLocaleScriptCode (null)
NSLocaleVariantCode (null)
NSLocaleCountryCode US

2. English (Region: China)

NSLocaleIdentifier en_CN
NSLocaleLanguageCode en
NSLocaleScriptCode (null)
NSLocaleVariantCode (null)
NSLocaleCountryCode CN

3. Simplified Chinese (Region: China)

NSLocaleIdentifier zh_CN
NSLocaleLanguageCode zh
NSLocaleScriptCode (null)
NSLocaleVariantCode (null)
NSLocaleCountryCode CN

4. Traditional Chinese (Region: China)

NSLocaleIdentifier zh-Hant_CN
NSLocaleLanguageCode zh
NSLocaleScriptCode Hant
NSLocaleVariantCode (null)
NSLocaleCountryCode CN

5. Simplified Chinese (Region: Unites States)

NSLocaleIdentifier zh-Hans_US
NSLocaleLanguageCode zh
NSLocaleScriptCode Hans
NSLocaleVariantCode (null)
NSLocaleCountryCode US

6. Traditional Chinese (Region: Unites States)

NSLocaleIdentifier zh-Hant_US
NSLocaleLanguageCode zh
NSLocaleScriptCode Hant
NSLocaleVariantCode (null)
NSLocaleCountryCode US
```

As we can see in the example "3. Simplified Chinese (Region: China)" `NSLocaleScriptCode` is nil. While in 5 it's there. On another hand for Traditional Chinese it looks like it always has `NSLocaleScriptCode` equal `Hant`. So I think we can do this:

```objective-c
    } else if ([languageCode isEqualToString:@"zh"]) {
        NSString *languageScript = [self.locale objectForKey:NSLocaleScriptCode];
        if ([languageScript isEqualToString:@"Hant"] == NO) { // Means Simplified
            return [self zhCNRelativeDateStringForComponents:components];
        }
    }
```

File: FormatterKit/TTTTimeIntervalFormatter.m [hunk 3]
```diff
@@ -584,6 +589,38 @@ - (NSString *)huRelativeDateStringForComponents:(NSDateComponents *)components
     return nil;
 }
 
+- (NSString *)zhCNRelativeDateStringForComponents:(NSDateComponents *)components {
```
Comment (yas375) [inline FormatterKit/TTTTimeIntervalFormatter.m]: If this is specific for Simplified Chinese, then let's call it `zhHansRelativeDateStringForComponents`.

File: FormatterKit/TTTTimeIntervalFormatter.m (hunk 2 above)
Comment (Elethom) [inline FormatterKit/TTTTimeIntervalFormatter.m]: Does it simply drop "Hans" when you set "zh-Hans_CN"?

By the way I didn't change anything in the Traditional Chinese part.

Comment (yas375) [inline FormatterKit/TTTTimeIntervalFormatter.m]: > Does it simply drop "Hans" when you set "zh-Hans_CN"?

This seems it be a little weird behavior. When I set region to China then it drops (see 3 from my previous comment), but if I use region US, but the language is still Simplified Chinese, then it's there (see 5).

> By the way I didn't change anything in the Traditional Chinese part.

I understand. The problem was that `zhCNRelativeDateStringForComponents` was called for both Simplified and Traditional.

**Comment by yas375** [comment]
@Elethom thanks a lot 👍

File: FormatterKit/TTTTimeIntervalFormatter.m (hunk 2 above)
Comment (Elethom) [inline FormatterKit/TTTTimeIntervalFormatter.m]: Understood. That missing "Hans" is probably some sort of bug in `NSLocale` I guess.

Comment (yas375) [inline FormatterKit/TTTTimeIntervalFormatter.m]: Yeah, looks like it is.

**Pull request merged**